.github/workflows/aod-iam.yml

```diff
@@ -0,0 +1,51 @@
+# Workflow that validates and handles AOD IAM request.
+name: 'aod-iam'
+
+on:
+  pull_request:
+    types: ['opened']
+    branches:
+      - 'main'
+  pull_request_review:
+    types: ['submitted']
+
+jobs:
+  # Check if it is an AOD request
+  is_iam:
+    uses: 'abcxyz/aod-tempate/.github/workflows/reusable-checker.yml@main' # Since it does not have any access to github context, directly ping the tag/ref here.
+    with:
+      path: 'iam.yaml'
+
+  # Add always fail job.
+  always-fail:
+    needs: 'is_iam'
+    if: ${{ needs.is_iam.outputs.is_aod == 'true'}}
+    runs-on: ubuntu-latest
+    steps:
+      - id: 'fail'
+        run: exit 1
+
+  set_inputs:
+    needs: 'is_iam'
+    if: ${{ needs.is_iam.outputs.is_aod == 'true'}}
+    uses: 'abcxyz/aod-tempate/.github/workflows/reusable-set-inputs.yml@main'
+
+  validate:
+    needs: 'set_inputs'
+    uses: 'abcxyz/access-on-demand/.github/workflows/iam-validate.yml@main' # Since it does not have any access to github context, directly ping the tag/ref here.
```

Comment:
I now wonder if we should merge `iam-validate` into `iam-handle`? Why we need two workflows?

Reply:
At first, since they will be in different occasions. we will always want to validate when pull request opened and approved. But we can combine them in one, this condition inside the workflow.

Reply:
Ignore this if we want to have a separate validation workflow on every PR commit